### PR TITLE
[REF432] packaging wheel for quicknxs

### PR DESCRIPTION
**Pull request by KedoKudo**
- Original Gitlab Enabler: [REF432](https://code.ornl.gov/sns-hfir-scse/reflectometry/reflectometry/-/issues/432)

This PR introduces the following changes:

- development environment file is now consolidated into `environment.yml`, runtime dependencies is defined in `setup.py`
- auto versioning tool `versioninggit` is added to build toolchain
- remove deprecated dependency files
- updated release note and readme

### CODE REVIEW

File: .github/workflows/unittest.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      run:
+        shell: bash -l {0}
+    steps:
+    - uses: actions/checkout@v2
+    - uses: conda-incubator/setup-miniconda@v2
+      with:
+        auto-update-conda: true
+        mamba-version: "*"
+        environment-file: environment.yml
+    - name: Install quicknxs via setup.py
```
Comment (peterfpeterson) [inline .github/workflows/unittest.yml:22]: I think the original instructions were a bit off. This job should actually be
```sh
python setup.py bdist_wheel
check-wheel-contents dist/*.whl
```
the python `build` module apparently doesn't know how to use `setup.py`

**Review by peterfpeterson — CHANGES_REQUESTED** [review]

**Comment by peterfpeterson** [comment]
The contents of the current rpm are 
```
/usr/bin/RefRedM
/usr/bin/quicknxs2
/usr/lib/python2.7/site-packages/reflectivity_ui-2.1.0-py2.7.egg-info
/usr/lib/python2.7/site-packages/reflectivity_ui-2.1.0-py2.7.egg-info/PKG-INFO
/usr/lib/python2.7/site-packages/reflectivity_ui-2.1.0-py2.7.egg-info/SOURCES.txt
/usr/lib/python2.7/site-packages/reflectivity_ui-2.1.0-py2.7.egg-info/dependency_links.txt
/usr/lib/python2.7/site-packages/reflectivity_ui-2.1.0-py2.7.egg-info/not-zip-safe
/usr/lib/python2.7/site-packages/reflectivity_ui-2.1.0-py2.7.egg-info/requires.txt
/usr/lib/python2.7/site-packages/reflectivity_ui-2.1.0-py2.7.egg-info/top_level.txt
/usr/lib/python2.7/site-packages/reflectivity_ui/__init__.py
/usr/lib/python2.7/site-packages/reflectivity_ui/__init__.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/__init__.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/config/__init__.py
/usr/lib/python2.7/site-packages/reflectivity_ui/config/__init__.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/config/__init__.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/config/gui.py
/usr/lib/python2.7/site-packages/reflectivity_ui/config/gui.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/config/gui.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/config/plotting.py
/usr/lib/python2.7/site-packages/reflectivity_ui/config/plotting.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/config/plotting.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/__init__.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/__init__.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/__init__.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/configuration.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/configuration.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/configuration.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/__init__.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/__init__.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/__init__.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/data_info.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/data_info.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/data_info.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/data_manipulation.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/data_manipulation.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/data_manipulation.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/data_set.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/data_set.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/data_set.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/genx_templates/polarized.gx
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/genx_templates/spinflip.gx
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/genx_templates/unpolarized.gx
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/gisans.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/gisans.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/gisans.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/instrument.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/instrument.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/instrument.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/off_specular.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/off_specular.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/off_specular.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/peak_finding.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/peak_finding.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/peak_finding.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/processing_workflow.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/processing_workflow.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/processing_workflow.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/quicknxs_io.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/quicknxs_io.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_handling/quicknxs_io.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_manager.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_manager.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/data_manager.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/event_handlers/__init__.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/event_handlers/__init__.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/event_handlers/__init__.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/event_handlers/main_handler.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/event_handlers/main_handler.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/event_handlers/main_handler.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/event_handlers/plot_handler.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/event_handlers/plot_handler.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/event_handlers/plot_handler.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/event_handlers/progress_reporter.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/event_handlers/progress_reporter.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/event_handlers/progress_reporter.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/__init__.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/__init__.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/__init__.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/compare_plots.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/compare_plots.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/compare_plots.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/icons_rc.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/icons_rc.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/icons_rc.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/mplwidget.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/mplwidget.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/mplwidget.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/ui_compare_widget.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/ui_compare_widget.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/ui_compare_widget.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/ui_main_window.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/ui_main_window.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/ui_main_window.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/ui_reduce_dialog.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/ui_reduce_dialog.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/ui_reduce_dialog.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/ui_result_viewer.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/ui_result_viewer.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/ui_result_viewer.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/ui_smooth_dialog.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/ui_smooth_dialog.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/generated/ui_smooth_dialog.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/main_window.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/main_window.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/main_window.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/plotting.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/plotting.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/plotting.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/reduction_dialog.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/reduction_dialog.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/reduction_dialog.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/result_viewer.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/result_viewer.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/result_viewer.pyo
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/smooth_dialog.py
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/smooth_dialog.pyc
/usr/lib/python2.7/site-packages/reflectivity_ui/interfaces/smooth_dialog.pyo
```
There must be a "good" way to compare that list to what is in this wheel

File: .github/workflows/unittest.yml (hunk 1 above)
Comment (KedoKudo) [reply .github/workflows/unittest.yml:22]: updated with the suggested edits

**Review by peterfpeterson — APPROVED** [review]
I've compared the contents with those of the v2.1 rpm and it looks good now.

**Pull request merged**